Two build issues on Mac M1 chip

## Bug Report

<!-- Thanks for your bug report! Don't worry if you can't fill out all the sections. -->

1. On MacOS, if `coreutils/gnubin` is installed, the `uname -p` will show `arm64`, neither `aarch64` nor `arm`, causing SSE to be enabled. ([checked here](https://github.com/tikv/tikv/blob/032a990aaf6ef0167bab2f466e00c42436b28b97/Makefile#L64))

```bash
/opt/homebrew/opt/coreutils/libexec/gnubin/uname -p
# output arm64
```

Note that this might be common if someone is also developing `kubernetes`, since `kubernetes` build guide requires `coreutils` to be installed.

2. The build guide mentioned `llvm` and `clang` should be installed. What's the correct way of installing them on M1 MacOS? I installed using `Homebrew` but the `Homebrew` version causes failed `openssl` build (See [this issue](https://github.com/openssl/openssl/issues/10709, basically, the `Homebrew` version llvm cannot recognize the `-c` parameter of `ranlib`).

ps. I can build successfully after uninstalling the homebrew version llvm.

```bash
...
  install libcrypto.a -> /Users/dongjin/code/tikv/target/debug/build/openssl-sys-8948495dd3a11ccf/out/openssl-build/install/lib/libcrypto.a
  make[1]: Leaving directory '/Users/dongjin/code/tikv/target/debug/build/openssl-sys-8948495dd3a11ccf/out/openssl-build/build/src'

  --- stderr
  ar: warning: creating apps/libapps.a
  ranlib: error: Invalid option: '-c'
  ar: warning: creating libssl.a
  ranlib: error: Invalid option: '-c'
  ar: warning: creating libcrypto.a
  ranlib: error: Invalid option: '-c'
  ranlib: error: Invalid option: '-c'
  make[1]: *** [Makefile:327: install_dev] Error 1
  thread 'main' panicked at '


  Error installing OpenSSL:
      Command: "make" "install_dev"
      Exit status: exit status: 2


      ', /Users/dongjin/.cargo/registry/src/github.com-1ecc6299db9ec823/openssl-src-111.15.0+1.1.1k/src/lib.rs:471:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
make: *** [Makefile:154: build] Error 101
```

### What version of TiKV are you using?
<!-- You can run `tikv-server --version` -->
Build from source (master)

### What operating system and CPU are you using?
<!-- If you're using Linux, you can run `cat /proc/cpuinfo` -->
Mac OS Monterey, M1.

### Steps to reproduce
<!-- If possible, provide a recipe for reproducing the error. A complete runnable program is good. -->


### What did you expect?

### What did happened?

--- a/Makefile
+++ b/Makefile
@@ -68,6 +68,9 @@ endif
 ifeq ($(shell uname -p),arm)
 ROCKSDB_SYS_SSE=0
 endif
+ifeq ($(shell uname -p),arm64)
+ROCKSDB_SYS_SSE=0
+endif
 
 # Build portable binary by default unless disable explicitly
 ifneq ($(ROCKSDB_SYS_PORTABLE),0)

